core/registration.c

```diff
@@ -211,6 +225,10 @@ static void prv_handleRegistrationReply(lwm2m_transaction_t * transacP,
 
             LOG("Registration successful");
         }
+        else if (packet != NULL && packet->code == COAP_231_CONTINUE)
```

Comment:
There is the potential that COAP_201_CREATED would come back before the final block is sent. In that case, the location should be stored, but the targetP->status not change to STATE_REGISTERED until the response to the final block.

Reply:
In the case where the COAP_201_CREATED comes back before the final block, the final block would have a COAP_231_CONTINUE response code. In this case there are no more blocks to send and targetP->status should be set to STATE_REGISTERED.